### PR TITLE
TF-2304 Implement sorting search result for mobile

### PR DESCRIPTION
**Pull request by hieutbui**
## Issue
---
#2304 

## Resolved
---
- Web responsive:

https://github.com/linagora/tmail-flutter/assets/80142234/58762c6b-73b0-41d2-9e11-f8c6264172ff

- Android:

https://github.com/linagora/tmail-flutter/assets/80142234/f705bd84-05fb-4231-8f53-92dad18a286c

### CODE REVIEW

File: lib/features/search/email/presentation/search_email_view.dart [hunk 1]
```diff
@@ -294,6 +321,53 @@ class SearchEmailView extends GetWidget<SearchEmailController>
...
+    return EmailSortOrderType.values
+      .map(
+        (sortType) => (
+          EmailSortByCupertinoActionSheetActionBuilder(
+            sortType.getTitle(context),
+            sortType,
+            sortTypeCurrent: sortOrderSelected,
+            iconLeftPadding: _responsiveUtils.isMobile(context)
+              ? const EdgeInsets.only(left: 12, right: 16)
+              : const EdgeInsets.only(right: 12),
```
Comment (dab246) [inline lib/features/search/email/presentation/search_email_view.dart:355]: Should be use `EdgeInsetsDirectinal` to support RTL mode

File: lib/features/search/email/presentation/search_email_view.dart [hunk 2]
```diff
@@ -294,6 +321,53 @@ class SearchEmailView extends GetWidget<SearchEmailController>
...
+            iconRightPadding: _responsiveUtils.isMobile(context)
+              ? const EdgeInsets.only(right: 12)
```
Comment (dab246) [inline lib/features/search/email/presentation/search_email_view.dart:357]: same

File: lib/features/search/email/presentation/widgets/email_sort_by_action_tile_widget.dart [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import 'package:core/core.dart';
```
Comment (dab246) [inline lib/features/search/email/presentation/widgets/email_sort_by_action_tile_widget.dart:1]: Import only the necessary objects

**Comment by hoangdat** [comment]
Screenshot 2023-11-15 at 17 47 1 src=

@hieutbui, weird result

**Comment by dab246** [comment]
> Screenshot 2023-11-15 at 17 47 12
> @hieutbui, weird result

@hieutbui  Please update demo again

**Comment by hieutbui** [comment]
> > Screenshot 2023-11-15 at 17 47 12
> > @hieutbui, weird result
> 
> @hieutbui Please update demo again

I updated the video demo for it.

**Comment by dab246** [comment]
@hieutbui Do we still have a problem here?

**Comment by hieutbui** [comment]
> @hieutbui Do we still have a problem here?

No, problem now is on the back-end side.

**Comment by dab246** [comment]
> > @hieutbui Do we still have a problem here?
> 
> No, problem now is on the back-end side.

Oke. good

**Review by dab246 — APPROVED** [review]

**Pull request merged**